### ISSUE TITLE
Microsoft accounts for Hotmail, Live.com, Office365 und Web-Outlook no longer working / supported - OAUTH2 support needed

### ISSUE THREAD

**Issue by io2345** [iobroker-community-adapters/ioBroker.email#261]
Microsoft will enforce the security of it's Email-Products, it will not be possible to use simply Username+Password after 16.September 2024. See here: [https://techcommunity.microsoft.com/t5/outlook-blog/keeping-our-outlook-personal-email-users-safe-reinforcing-our/ba-p/4164184](url)
Is there anything planned for the email-plugin of ioBroker? Or can these accounts no longer be used then?

**Comment by Apollon77** [comment]
let's phrase it that way ... As long as microsoft mot tell exact details on what and how it needs to be done and which "new" more secure options they apple it is hard to tell anything. But likely yes it will not work until someone adjusted it to whatever microsoft uses and not tell before ... grmpff

**Comment by mcm1957** [comment]
Due to the very short timeframe MS$ used to annoucne this, do not assume that an adaption will be done until Sepztember.
If you know a detailled specification how an access could be done to MS$ servers please provide a link.

**Comment by mcm1957** [comment]
> let's phrase it that way ... As long as microsoft mot tell exact details on what and how it needs to be done and which "new" more secure options they apple it is hard to tell anything. But likely yes it will not work until someone adjusted it to whatever microsoft uses and not tell before ... grmpff

Sieht so aus als würe MS$ OAUTH verlangen:
https://support.microsoft.com/de-de/office/moderne-authentifizierungsmethoden-sind-jetzt-erforderlich-um-die-synchronisierung-von-outlook-e-mails-in-nicht-microsoft-e-mail-apps-fortzusetzen-c5d65390-9676-4763-b41f-d7986499a90d

**Comment by mcm1957** [comment]
nodemailer untersützt scheinbar oauth

https://github.com/nodemailer/nodemailer/blob/master/examples/oauth2.js

**Comment by mcm1957** [comment]
https://forum.iobroker.net/topic/77123/adapter-e-mails-senden-geht-nicht-mehr

**Comment by tecfreakz** [comment]
Seit kurzem scheint Outlook/Hotmail/M365 nicht mehr mit diesem Adapter zu funktionieren.

<html>
<body>

email.0 | 2024-10-06 15:18:32.686 | error | Error 535 5.7.139 Authentication unsuccessful, basic authentication is disabled. [FR4P281CA0420.DEUP281.PROD.OUTLOOK.COM 2024-10-06T13:18:32.682Z 08DCE51CB70F88C4]
-- | -- | -- | --



</body>
</html>

Gibt's Ideen, wie man das wieder zum Laufen bekommt?

**Comment by io2345** [comment]
Just for understanding: If this adapter is based on nodemailer, and nodemailer supports oauth - can it be so hard to get things up and running again (for microsoft-based accounts like hotmail)?

**Comment by mcm1957** [comment]
Feel free to provide a PR. It would be welcome for sure.

**Comment by mcm1957** [comment]
GMAIL requires oauth starting with 1.1.2025 too if the follwoing link is correct
https://support.google.com/a/answer/14114704?sjid=3966942068221268267-EU

**Comment by UncleSamSwiss** [comment]
> Just for my understanding: If this adapter is based on nodemailer, and nodemailer supports oauth - can it really be so hard to get things up and running again (for microsoft-based accounts like hotmail)?

AFAIK you need a client ID and a client secret for OAuth. We can (for obvious reasons) not provide these in the adapter. So, I'm not sure how we can get this working for everybody, unless users are ready to create these credentials themselves.

**Comment by tarikweiss** [comment]
Alright, just checked the transition timeline of Microsoft. It is *extraordinary* confusing. In the german article you get a suggestion how to use the service with "Security Cameras" which are just supporting the normal SMTP login. In the english version of the article there is just the first sentence, which is indicating such use case. Furthermore there is this very definitive article about the transition time https://support.microsoft.com/en-us/office/modern-authentication-methods-now-needed-to-continue-syncing-outlook-email-in-non-microsoft-email-apps-c5d65390-9676-4763-b41f-d7986499a90d . And then there is the microsoft account stating, that app-passwords still supported, but not specifying their exact use cases (except Xbox360). After all, there is just the way to switch to OAuth. The further consequences have been explained already.

**Comment by Wondermusic** [comment]
After some problems with my old mail provider, I tried also Office365.
I have no problems using this service.

On the other side i have problems with gmail & gmx...

**Comment by mcm1957** [comment]
fixed with 2.0.0

**Comment by io2345** [comment]
> fixed with 2.0.0

Best news today, great! When will it be available to the public? My ioBroker installation does not offer a newer version than 1.3.1 until now

**Comment by mcm1957** [comment]
Release 2.0.0 is alteady available at BETA/LATEST repository.

**Comment by GermanBluefox** [comment]
> I installed the version 2.0.2 manually from ioBroker ("install dedicated version" -> "enter version manually". Had to pull up admin to version 7.6.2 as well). Is there a guide available on how to get this OAuth2 stuff done? What do I have to do?

Honestly, I thought that this was self-explanatory.

![Image](https://github.com/user-attachments/assets/8c29bdbd-1c17-48a8-9f58-a51308d4ae4e)

After the button pressed, the new Tab should be opened, if not, allow new tabs and then just enter your login, password. After you entered login/password the oauth2 token will be passed to ioBroker automatically and the new tab will close itself

**Comment by io2345** [comment]
Ah, verstehe, man muss auf den Service "Office365" umstellen. Bei mir war natürlich noch "Hotmail" drin.
Man muss anschließend in den Instanzeinstellungen zuerst unten auf "Speichern" drücken, bevor man den "Test" Knopf drückt.
Vielen Dank!

**Comment by GabeHH78** [comment]
Hi,

I do this requested way and get also green token sign, but if I sent the test mail, I still get follow error message

Error: 535 5.7.139 Authentication unsuccessful, basic authentication is disabled. [FR3P281CA0117.DEUP281.PROD.OUTLOOK.COM 2025-03-21T07:45:40.831Z 08DD6490F68C6FCF]

Someone has a idea how I can do it?

Thanks

**Comment by io2345** [comment]
That's the message I got as well, until I changed "service" in the Adapter's configuration to "Office365". Did you do that and press the "Save" button after that?

**Comment by GabeHH78** [comment]
yes just try it again change it to office365 and get still the same error. For the oauth he show me it in green and that token is valid

**Comment by tarikweiss** [comment]
The point is, that you need to enable external authentication in your Outlook Accounts. Do you use private mail addresses or Azure AD / Microsoft 365 Accounts? If the latter then you need to configure this in the Microsoft Admin Center.

**Comment by GabeHH78** [comment]
thanks i found the mistake

**Comment by io2345** [comment]
> thanks i found the mistake

Please do people a favour and write as exactly as possible, what the error was and how you could fix it.
This is the main purpose of help threads: Find help for yourself and enable others to find their errors themselves.
Thank you.

**Comment by mcm1957** [comment]
In general I suggest to open a Issue at the FORUM (forum.iobroker.net) for usage discussion / help. As far as I see there is NO know bug currently at the code. So this isseu will be closes as soon as the BETA version can be published to stable. This requires an update ob admin at stable too.

**Comment by io2345** [comment]
> In general I suggest to open a Issue at the FORUM (forum.iobroker.net) for usage discussion / help. As far as I see there is NO know bug currently at the code. So this isseu will be closes as soon as the BETA version can be published to stable. This requires an update ob admin at stable too.

Thank you for your efforts. You are absolutely right.
I opened a thread on the mentioned forum some time ago. The link is [https://forum.iobroker.net/topic/75945/wie-microsoft-mails-versenden-nach-16-september/6](url)

**Comment by GermanBluefox** [comment]
Can I close it?

**Comment by io2345** [comment]
I'd vote for "Yes"

**Comment by mcm1957** [comment]
Implemented with 2.0.2
